clients/ts/signalr/tests/HubConnection.test.ts

```diff
@@ -48,6 +48,46 @@ describe("HubConnection", () => {
         });
     });
 
+    describe("ping", () => {
+        it("automatically sends multiple pings", async () => {
+            const connection = new TestConnection();
+            const hubConnection = createHubConnection(connection);
+
+            hubConnection.pingIntervalInMilliseconds = 150;
+
+            try {
+                await hubConnection.start();
+                await delay(322);
+
+                const numPings = connection.sentData.filter((s) => JSON.parse(s).type === MessageType.Ping).length;
+                expect(numPings).toBeGreaterThanOrEqual(2);
+            } finally {
+                await hubConnection.stop();
+            }
+        });
+
+        it("doesn't ping while sending other messages", async () => {
+            const connection = new TestConnection();
+            const hubConnection = createHubConnection(connection);
+
+            hubConnection.pingIntervalInMilliseconds = 150;
+
+            try {
+                await hubConnection.start();
+
+                for (let i = 0; i < 30; i++) {
+                    hubConnection.send("some message");
```

Comment:
This should be `await`ed. Because of the `delay` it should be fine, but it's good hygiene to make sure we `await` it.